packages/csv-parser-state/src/main.js

```diff
@@ -0,0 +1,190 @@
+        this.logger.error(`At row: ${this._rowIndex}, ${error}`)
+        output.emit('error', error)
+      })
+      .doto(() => {
+        this._rowIndex += 1
+        this.logger.debug(`At row: ${this._rowIndex}, Successfully parsed`)
+      })
+      .pipe(JSONStream.stringify())
+      .pipe(output)
+  }
+
+  async _transformTransitions({
+    transitions,
+    ...restOfState
+  }: StateWithUnresolvedTransitions) {
+    if (!transitions) return restOfState
+    // We setup the client here because it is unnecessary
+    // if there are no transitions
+    this._setupClient()
+    const stateRequests = transitions.map(transitionState =>
+      this._buildStateRequest(transitionState)
+    )
+    const resolvedStates = await Promise.all(stateRequests)
+    const stateReferences = resolvedStates.map(response => ({
+      typeId: 'state',
+      id: response.body.id,
+    }))
+    return { ...restOfState, transitions: stateReferences }
+  }
+
+  _buildStateRequest(stateKey: string): Promise<SuccessResult> {
+    const stateService = this._createStateService()
+    const uri = stateService.byKey(stateKey).build()
+    return this._fetchStates(uri)
+  }
+
+  _setupClient() {
+    if (!this.apiConfig)
+      throw new Error('The constructor must be passed an `apiConfig` object')
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareWithExistingToken(
+          this.accessToken ? `Bearer ${this.accessToken}` : ''
+        ),
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+  }
+
+  _createStateService() {
+    return createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    }).states
+  }
+
+  _mapMultiValueFieldsToArray({
+    roles,
+    transitions,
+    ...restOfState
+  }: StateWithStringTransitions) {
+    const mappedObject = {}
+    if (transitions)
+      mappedObject.transitions = this._mapStringToArray(transitions)
+    if (roles) mappedObject.roles = this._mapStringToArray(roles)
+    return { ...mappedObject, ...restOfState }
+  }
+
+  _mapStringToArray(value: string): Array<string> {
+    return value.split(this.csvConfig.multiValueDelimiter)
+  }
+
+  // Highlang signature for custom error handling
+  _handleErrors(error: any, callback: Function) {
+    this._rowIndex += 1
+    if (this.continueOnProblems)
+      // Log warning and continue
+      this.logger.warn(`Ignoring error at row: ${this._rowIndex}, ${error}`)
+    else
+      // <- Rethrow the error to `.stopOnError()`
+      callback(error)
+  }
+
+  // Remove fields with empty values from the state objects
+  static _removeEmptyFields(item: Object): Object {
+    // "acc" is "state" but lint doesn't like mutations
+    return Object.entries(item).reduce((acc, entry) => {
```

Comment:
You could destructure the item inline herr already to not have to refer to it by deferring it below.